src/shared_modules/indexer_connector/src/indexerConnector.cpp

```diff
@@ -292,36 +291,44 @@ void IndexerConnector::initialize(const nlohmann::json& templateData,
     };
 
     // Define the success callback
-    auto onSuccess = [](const std::string&)
+    nlohmann::json templateData {};
+    auto onSuccess = [&templateData](const std::string& response)
     {
-        // Not used
+        try
+        {
+            templateData =
+                nlohmann::json::parse(response).at("index_templates").front().at("index_template").at("template");
+        }
+        catch (const nlohmann::json::exception& e)
+        {
+            logDebug1(IC_NAME, "Failed to parse index template: %s", e.what());
+            throw std::runtime_error("Failed to parse index template.");
+        }
     };
 
-    // Initialize template.
-    HTTPRequest::instance().put(HttpURL(selector->getNext() + "/_index_template/" + m_indexName + "_template"),
-                                templateData,
-                                onSuccess,
-                                onError,
-                                "",
-                                DEFAULT_HEADERS,
-                                secureCommunication);
-
     // Initialize Index.
-    HTTPRequest::instance().put(HttpURL(selector->getNext() + "/" + m_indexName),
-                                templateData.at("template"),
+    HTTPRequest::instance().get(HttpURL(selector->getNext() + "/_index_template/" + m_indexName),
```

Comment:
This must also be removed. We do not initialize anything else here. Remove this method.

Reply:
Wrong approach, removed. Template and index are supposed to be initialized.